### PR TITLE
[terra-worklist-data-grid] Make cell selection honor prop values

### PR DESCRIPTION
**Pull request by cm9361**
### Summary

**What was changed:**
The WorklistDataGrid was updated to honor the selection state passed by consumers via props instead of attempting to manage the selection state for cells with it own internal state variables.

**Why it was changed:**
The change was made so that consumers had full control of the cell selection state and provides flexibility for implementations such as the MPages Worklist.

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details
N/A

**This PR resolves:**

UXPLATFORM-9497 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-worklist-data-grid/CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 # Changelog
 
 ## Unreleased
-
-* Added
+* Fixed
+  * Removed roving tabIndex and resolved focus management issues within cells.
+  * Reduced number of unnecessary rerenders in the `terra-worklist-data-grid` component.
```
Comment (sdadn) [inline packages/terra-worklist-data-grid/CHANGELOG.md:6]: This seems to be a merge conflict

Comment (cm9361) [reply packages/terra-worklist-data-grid/CHANGELOG.md]: I believe that I have resolved the merge conflict issue now.

**Review by sycombs — APPROVED** [review]

File: packages/terra-worklist-data-grid/CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 * Fixed
   * Reduced number of unnecessary rerenders in the `terra-worklist-data-grid` component.
+  * Updated cells to honor the value of props passed by consumers.
```
Comment (sdadn) [inline packages/terra-worklist-data-grid/CHANGELOG.md:14]: We should move this to the `changed` section.

**Review by sdadn — APPROVED** [review]
Have 1 comment but approving since that should be a quick fix

File: packages/terra-worklist-data-grid/CHANGELOG.md (hunk 2 above)
Comment (cm9361) [reply packages/terra-worklist-data-grid/CHANGELOG.md:14]: Well, this was kind of a defect, but I can move it if we feel that is best.

**Review by artpark — APPROVED** [review]

**Pull request merged**